src/DependencyInjection/DI/test/ServiceProviderContainerTests.cs

```diff
@@ -180,12 +188,19 @@ public void ScopeDispose_PreventsServiceResolution()
             Assert.NotNull(provider.CreateScope());
         }
 
-        [Theory(Skip = "We don't support value task services currently")]
+        [Theory]
         [InlineData(ServiceLifetime.Transient)]
         [InlineData(ServiceLifetime.Scoped)]
         [InlineData(ServiceLifetime.Singleton)]
         public void WorksWithStructServices(ServiceLifetime lifetime)
         {
+#if NETCOREAPP5_0
+            if (GetType() == typeof(ServiceProviderILEmitContainerTests) && lifetime == ServiceLifetime.Scoped)
```

Comment:
As mentioned in the pull request summary/description, this specific scenario fails with an exception:

```cs
System.InvalidProgramException : Common Language Runtime detected an invalid program.
   at ResolveService(ILEmitResolverBuilderRuntimeContext , ServiceProviderEngineScope )
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.Tests.ServiceProviderContainerTests.WorksWithStructServices(ServiceLifetime lifetime)
```

So until there's a solution for that, I've made it skip this specific scenario.

I'm very curious to know why this specific scenario fails **only** when running the unit tests targeting `netcoreapp5.0`. Perhaps there's a change in the CLR which has introduced a bug?

Reply:
@pakrym and @davidfowl I think the two of you, and perhaps @brandondahler would be interested in this.

Reply:
I don't believe it is a CLR bug, I would presume it is that the newer runtime is better flagging IL that has (presumably) undefined behavior*.  I say this is because with a simple correction in the IL being generated, I was able to make the test succeed.

See my comment below for a change that prevents this exception from being thrown.

_* I am not an authoritative or even particularly well-informed source, so take my opinion with a grain of salt._